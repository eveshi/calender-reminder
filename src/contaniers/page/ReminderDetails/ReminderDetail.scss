@import '../../../stylesheets/main.scss';

.ReminderDetails {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $color-primary-dark;

  &_header {
    width: 100%;
    height: 15vh;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    display: flex;
    justify-content: space-between;

    a {
      border-radius: 50%;
      background-color: $color-secondary-dark;
      height: 5rem;
      width: 5rem;
      position: relative;
      @include transitionSpeed(fast);

      svg {
        position: absolute;
        left: .5rem;
        top: .5rem;
      }

      &:hover {
        transform: scale(0.98);
      }
    }
  }

  &_content {
    height: 85vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: $color-secondary-dark;

    & > * {
      margin-bottom: 4rem;
    }

    &_time {
      font-size: 3.6rem;
      color: $color-secondary-dark;
    }

    &_reminder {
      width: 80%;
      font-size: 2.4rem;
      text-align: center;
      word-wrap: break-word;
    }

    .Button {
      &:hover {
        background-color: transparent;
      }

      svg {
        height: 4rem;
        width: 4rem;
        text-decoration: none;
        fill: $color-secondary-dark;

        &:hover {
          opacity: 0.7;
        }
      }
    }
  }
}