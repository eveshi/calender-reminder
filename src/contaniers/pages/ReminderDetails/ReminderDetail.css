*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.ReminderDetails {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #060505; }
  .ReminderDetails_header {
    width: 100%;
    height: 15vh;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    display: flex;
    justify-content: space-between; }
    .ReminderDetails_header a {
      border-radius: 50%;
      background-color: #BD924B;
      height: 5rem;
      width: 5rem;
      position: relative;
      transition: all .2s ease; }
      .ReminderDetails_header a .Button {
        background-color: transparent; }
      .ReminderDetails_header a svg {
        position: absolute;
        left: .5rem;
        top: .5rem; }
      .ReminderDetails_header a:hover {
        transform: scale(0.98); }
  .ReminderDetails_content {
    height: 85vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: #BD924B; }
    .ReminderDetails_content > * {
      margin-bottom: 4rem; }
    .ReminderDetails_content_time {
      font-size: 3.6rem;
      color: #BD924B; }
    .ReminderDetails_content_reminder {
      width: 80%;
      font-size: 2.4rem;
      text-align: center;
      word-wrap: break-word; }
    .ReminderDetails_content .Button:hover {
      background-color: transparent; }
    .ReminderDetails_content .Button svg {
      height: 4rem;
      width: 4rem;
      text-decoration: none;
      fill: #BD924B; }
      .ReminderDetails_content .Button svg:hover {
        opacity: 0.7; }
