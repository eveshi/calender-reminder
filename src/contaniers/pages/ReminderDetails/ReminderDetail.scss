@import '../../../stylesheets/main.scss';

.ReminderDetails {
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $color-primary-dark;

  &_header {
    width: 100%;
    height: 15vh;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-top: 2rem;
    display: flex;
    justify-content: space-between;

    @include link_button
  }

  &_content {
    height: 85vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: $color-secondary-dark;

    & > * {
      margin-bottom: 4rem;
    }

    &_time {
      font-size: 3.6rem;
      color: $color-secondary-dark;
    }

    &_reminder {
      width: 80%;
      font-size: 2.4rem;
      text-align: center;
      word-wrap: break-word;
    }

    .Button {
      &:hover {
        background-color: transparent;
      }

      svg {
        height: 4rem;
        width: 4rem;
        text-decoration: none;
        fill: $color-secondary-dark;

        &:hover {
          opacity: 0.7;
        }
      }
    }
  }
}