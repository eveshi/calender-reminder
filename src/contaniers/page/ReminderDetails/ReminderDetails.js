import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import dateFns from 'date-fns';
import { Link } from 'react-router-dom';

import { timeStringGenerator } from '../../../utility/utility';
import Button from '../../../components/Button/Button';
import { DELETE_REMINDER, CHANGE_SESSION } from '../../../store/actions/index';

import Close from '../../../assets/icon/Close';
import Edit from '../../../assets/icon/Edit';
import Delete from '../../../assets/icon/Delete';

import './ReminderDetail.css';

class ReminderDetails extends PureComponent {
  cleanSession = () => {
    const {
      changeSession,
    } = this.props;
    const reminder = {
      id: null,
      time: null,
      reminder: null,
    };
    changeSession(null, null, reminder);
  }

  deleteReminderHandler = () => {
    const {
      year,
      month,
      day,
      id,
      deleteReminder,
    } = this.props;

    const {
      cleanSession,
    } = this;

    const date = dateFns.format(new Date(year, month, day), 'YYYYMMDD');

    deleteReminder(date, id);
    cleanSession();
  }

  render() {
    const {
      year,
      month,
      day,
      time,
      reminder,
    } = this.props;

    const {
      deleteReminderHandler,
      cleanSession,
    } = this;

    const timeString = timeStringGenerator(time);

    return (
      <div className="ReminderDetails">
        <div className="ReminderDetails_header">
          <Link to="/">
            <Button onClick={() => cleanSession()}>
              <Close />
            </Button>
          </Link>
          <Link to="/edit_reminder">
            <Edit />
          </Link>
        </div>
        <div className="ReminderDetails_content">
          <p className="ReminderDetails_content_time">
            {year}
            -
            {month}
            -
            {day}
          </p>
          <p className="ReminderDetails_content_time">
            {timeString}
          </p>
          <p className="ReminderDetails_content_reminder">
            {reminder}
          </p>
          <Link to="/">
            <Button onClick={() => deleteReminderHandler()}>
              <Delete />
            </Button>
          </Link>
        </div>
      </div>
    );
  }
}

ReminderDetails.propTypes = {
  year: PropTypes.number.isRequired,
  month: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  time: PropTypes.number.isRequired,
  reminder: PropTypes.string.isRequired,
  deleteReminder: PropTypes.func.isRequired,
  changeSession: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  year: state.date.year,
  month: state.date.month,
  day: state.date.day,
  reminder: state.session.reminder.reminder,
  id: state.session.reminder.id,
  time: state.session.reminder.time,
});

const mapActionToProps = dispatch => ({
  deleteReminder: (date, id) => dispatch(DELETE_REMINDER(date, id)),
  changeSession: reminder => dispatch(CHANGE_SESSION(null, null, reminder)),
});

export default connect(mapStateToProps, mapActionToProps)(ReminderDetails);
