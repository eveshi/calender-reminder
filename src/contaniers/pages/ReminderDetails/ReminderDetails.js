import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import dateFns from 'date-fns';
import { Link } from 'react-router-dom';

import { timeStringGenerator, monthStringGenerator } from '../../../utility/utility';
import Button from '../../../components/Button/Button';
import * as actions from '../../../store/actions/index';

import Close from '../../../assets/icon/Close';
import Edit from '../../../assets/icon/Edit';
import Delete from '../../../assets/icon/Delete';

import './ReminderDetail.css';

export class ReminderDetails extends PureComponent {
  deleteReminderHandler = () => {
    const {
      year,
      month,
      day,
      id,
      deleteReminderHandler,
      cleanSession,
    } = this.props;

    const date = dateFns.format(new Date(year, month, day), 'YYYYMMDD');

    deleteReminderHandler(date, id);
    cleanSession();
  }

  render() {
    const {
      year,
      month,
      day,
      time,
      reminder,
    } = this.props;

    const {
      deleteReminderHandler,
      cleanSession,
    } = this;

    const timeString = timeStringGenerator(time);

    return (
      <div className="ReminderDetails">
        <div className="ReminderDetails_header">
          <Link to="/">
            <Button onClick={() => cleanSession()}>
              <Close />
            </Button>
          </Link>
          <Link to="/edit_reminder">
            <Edit />
          </Link>
        </div>
        <div className="ReminderDetails_content">
          <p className="ReminderDetails_content_time">
            {year}
            &nbsp;-&nbsp;
            {monthStringGenerator(month)}
            &nbsp;-&nbsp;
            {day}
          </p>
          <p className="ReminderDetails_content_time">
            {timeString}
          </p>
          <p className="ReminderDetails_content_reminder">
            {reminder}
          </p>
          <Link to="/">
            <Button onClick={() => deleteReminderHandler()}>
              <Delete />
            </Button>
          </Link>
        </div>
      </div>
    );
  }
}

ReminderDetails.propTypes = {
  year: PropTypes.number.isRequired,
  month: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  time: PropTypes.number.isRequired,
  reminder: PropTypes.string.isRequired,
  deleteReminderHandler: PropTypes.func.isRequired,
  cleanSession: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  year: state.reminderState.date.year,
  month: state.reminderState.date.month,
  day: state.reminderState.date.day,
  reminder: state.sessionState.reminder.reminder,
  id: state.sessionState.reminder.id,
  time: state.sessionState.reminder.time,
});

const mapActionToProps = dispatch => ({
  deleteReminderHandler: (date, id) => dispatch(actions.deleteReminder(date, id)),
  cleanSession: () => dispatch(actions.cleanSession()),
});

export default connect(mapStateToProps, mapActionToProps)(ReminderDetails);
