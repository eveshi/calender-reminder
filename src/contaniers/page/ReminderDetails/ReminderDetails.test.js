
import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ReminderDetails } from './ReminderDetails';

configure({ adapter: new Adapter() });

const props = {
  year: Math.random().toFixed(4)*4,
  month: Math.random().toFixed(2)*2,
  day: Math.random().toFixed(2)*2,
  id: '12345678',
  time: Math.random().toFixed(4)*4,
  reminder: 'abcdefg',
  deleteReminder: jest.fn(),
}

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ReminderDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});