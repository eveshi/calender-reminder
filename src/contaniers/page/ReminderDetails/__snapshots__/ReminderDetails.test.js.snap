// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div
  className="ReminderDetails"
>
  <div
    className="ReminderDetails_header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="Button"
        disabled={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="200"
          viewBox="0 0 1024 1024"
          width="200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="add_reminder"
          >
            DELETE
          </title>
          <path
            d="M809.984 274.005333l-237.994667 237.994667 237.994667 237.994667-59.989333 59.989333-237.994667-237.994667-237.994667 237.994667-59.989333-59.989333 237.994667-237.994667-237.994667-237.994667 59.989333-59.989333 237.994667 237.994667 237.994667-237.994667z"
          />
        </svg>
      </button>
    </a>
    <a
      href="/edit_reminder"
      onClick={[Function]}
    >
      <svg
        className="icon"
        height="200"
        viewBox="0 0 1024 1024"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="add_reminder"
        >
          ADD_REMINDER
        </title>
        <path
          d="M884.010667 299.989333l-77.994667 77.994667-160-160 77.994667-77.994667q11.989333-11.989333 29.994667-11.989333t29.994667 11.989333l100.010667 100.010667q11.989333 11.989333 11.989333 29.994667t-11.989333 29.994667zM128 736l472.021333-472.021333 160 160-472.021333 472.021333-160 0 0-160z"
        />
      </svg>
    </a>
  </div>
  <div
    className="ReminderDetails_content"
  >
    <p
      className="ReminderDetails_content_time"
    >
      2033
      -
      7
      -
      18
    </p>
    <p
      className="ReminderDetails_content_time"
    >
      12:00
    </p>
    <p
      className="ReminderDetails_content_reminder"
    >
      abcdefg
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="Button"
        disabled={null}
        id={null}
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon"
          height="200"
          viewBox="0 0 1024 1024"
          width="200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="add_reminder"
          >
            DELETE
          </title>
          <path
            d="M809.984 169.984l0 86.016-596.010667 0 0-86.016 148.010667 0 43.989333-41.984 212.010667 0 43.989333 41.984 148.010667 0zM256 809.984l0-512 512 0 0 512q0 34.005333-25.984 59.989333t-59.989333 25.984l-340.010667 0q-34.005333 0-59.989333-25.984t-25.984-59.989333z"
          />
        </svg>
      </button>
    </a>
  </div>
</div>
`;
